connectors/base/base.go

```diff
@@ -188,3 +190,8 @@ func (c *Connector) Shutdown() error {
 	}
 	return c.Next.Shutdown()
 }
+
+// Name returns the name of the connector
+func Name() string {
```

Comment:
In order to call `Name` on a connector, the connector has to be created, but the name is needed to create the connector which is why it's at the package level and not part of the interface.